PokemonRampUp/ViewController/DetailsViewController.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  DetailsViewController.swift
+//  PokemonRampUp
+//
+//  Created by Thushen Mohanarajah on 2023-03-07.
+//
+
+import UIKit
+
+class DetailsViewController: UIViewController {
+
+    
+    @IBOutlet weak var moveTableView: UITableView!
+    @IBOutlet weak var imageView: UIImageView!
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var typeLabel: UILabel!
+
+    var pokemon: Pokemon!
+    var pokemonMoves: [String]?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        imageView.image = pokemon.images.frontDefault
+        imageView.layer.cornerRadius = imageView.frame.size.width / 2
+        imageView.clipsToBounds = true
+        nameLabel.text = pokemon.name
+        typeLabel.text = pokemon.elementTypes[0].name
+        
+        moveTableView.delegate = self
+        moveTableView.dataSource = self
+        setPokemonMoveList()
+    }
+    
+    func setPokemonMoveList(){
+        self.pokemonMoves = []
+        
+        for move in pokemon.moves{
+            self.pokemonMoves?.append(move.name)
+        }
+        moveTableView.reloadData()
```

Comment:
There is another way to do the same thing here.
Suggestion: 

```
self.pokemonMoves = pokemon.moves.compactMap { $0.name }
moveTableView.reloadData()
```

Reply:
I added it

Reply:
I have one question tho. "$0" is basically the current element being processed right? 

```
for move in pokemon.moves{
        self.pokemonMoves?.append(move.name)
 }
  ```
In the code spinet above the name of the current element being processed it move